### PR TITLE
Unit testing of single multiple answer question

### PR DESCRIPTION
**Pull request by viharsanghavi**
#**Fixed Issues**

#21192

**Files Changed**

Bootstrap.cs - Add mapper initialization
QuestionRepositoryTest.cs - Add test method for AddSingleMultipleAnswerQuestion
Startup.cs - Remove extra mapper initialization
questions-single-multiple-answer.html - Solve click button event issue
questions-single-multiple-answer.component.ts - Add method to add option
QuestionRepository.cs -Add application user
IQuestionRepository.cs - Add application user

**Checks**

- [X] Naming Conventions
- [X] Server side code comments (proper English, grammar and no spelling mistakes)
- [X] Client side code comments (proper English, grammar and no spelling mistakes)
- [X] Unused variables, methods, blank spaces and name spaces. 
- [X] Optimal Code and code formatting
- [X] Commit History is proper, no merge is done. 
- [X] Commit/pull request messages should be proper to justify your feature
- [ ] All test cases are executed provided by tester.

**Known Issues**
N/A

**Comments**

anything that you want to convey about this pull request

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Test/Questions/QuestionsRepositoryTest.cs [hunk 1]
```diff
@@ -13,15 +21,145 @@ namespace Promact.Trappist.Test.Questions
     public class QuestionsRepositoryTest : BaseTest
     {
         private readonly IQuestionRespository _questionRepository;
+        private readonly Mock<IHostingEnvironment> _hostingEnvironmentMock;
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Test/Questions/QuestionsRepositoryTest.cs]: MAJOR Remove the "_hostingEnvironmentMock" field and declare it as a local variable in the relevant methods. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1450)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs [hunk 2]
```diff
@@ -37,11 +35,11 @@ public QuestionRepository(TrappistDbContext dbContext, UserManager<ApplicationUs
         /// <param name="questionAC">Object of QuestionAC</param>
         /// <param name="userEmail">Email id of user</param>
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: comment is not matching with a parameter.

File: Trappist/src/Promact.Trappist.Test/Questions/QuestionsRepositoryTest.cs [hunk 3]
```diff
@@ -13,15 +21,143 @@ namespace Promact.Trappist.Test.Questions
...
+        }
+
+        /// <summary>
+        /// Test to add multiple answer Question
+        /// </summary>
+        [Fact]
+        public async Task AddMultipleAnswerQuestionAsync()
+        {
+            var multipleAnswerQuestion = await CreateMultipleAnswerQuestion();
+            ApplicationUser user = new ApplicationUser() { Email = "vihar@promactinfo.com", UserName = "vihar@promactinfo.com" };
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Test/Questions/QuestionsRepositoryTest.cs]: the magic string should be used only once. if same magic string occurs more than once then please use a variable for it.

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-single-multiple-answer/questions-single-multiple-answer.component.ts [hunk 4]
```diff
@@ -70,6 +70,21 @@ export class SingleMultipleAnswerQuestionComponent {
         if (this.noOfOptionShown === 2) {
             this.isClose = true;
         }
+        if (this.value == optionIndex) {
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-single-multiple-answer/questions-single-multiple-answer.component.ts]: ===

File: Trappist/src/Promact.Trappist.Test/Questions/QuestionsRepositoryTest.cs [hunk 5]
```diff
@@ -13,15 +17,143 @@ namespace Promact.Trappist.Test.Questions
     public class QuestionsRepositoryTest : BaseTest
     {
         private readonly IQuestionRespository _questionRepository;
+        private readonly UserManager<ApplicationUser> _userManager;
+        ApplicationUser user;
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Test/Questions/QuestionsRepositoryTest.cs]: MAJOR Make "user" "readonly". [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS2933)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request merged**